eng/pipelines/client.yml

```diff
@@ -164,27 +164,27 @@ jobs:
     strategy:
       matrix:
         Linux - Java 8:
-          OSName: 'Linux'
+          Platform: 'Linux'
```

Comment:
Can we do away with this completely and just use the OSVmImage instead?

Reply:
I personally use in in the python repo with stuff like test run naming and the like. It's a nice vanity thing. 

I also think it allows us to simplify some of the `yml if` checks by quite a bit.

Reply:
Actually stand corrected. Looks like the code that actually uses platform is commented out. Will inquire with Mitch/Jim.

Reply:
As for the conditions we could also consider startsWith https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#startswith.

Reply:
Another thing to keep in mind is that I think `Platform` is used as a env variable in other context and might potentially cause some issues. See example https://stackoverflow.com/questions/2507856/why-is-my-platform-environment-variable-defined-as-bnb, I'm not sure if it will impact us here but it could potentially in other repo's.

Reply:
I feel like we'd be better off with @weshaggard's first suggestion of something like OSVmImage, (something we know isn't going to tangle with existing variables and would be less likely to be chosen as a new variable). I like that better than "platform".

Reply:
Ok. I'll completely rip out the name in favor of a `startswith` against the osvmimage.

@chidozieononiwu , you said you were looking at the OS check python script right? What are you thinking? Just going to make it do a `<argument>.lower().startswith(<os.platform>)`?